File: bundles/gitlab-eclipse-plugin/src/com/gitlab/eclipse/lsp/InlineCompletionParams.java
package com.gitlab.eclipse.lsp;

import org.eclipse.lsp4j.Position;
import org.eclipse.lsp4j.TextDocumentIdentifier;

/** LSP 3.18 textDocument/inlineCompletion request params (subset we need). */
public record InlineCompletionParams(TextDocumentIdentifier textDocument, Position position,
		InlineCompletionContext context) {}

File: src/main/kotlin/com/gitlab/eclipse/lsp/messages/InlineCompletionParams.kt
package com.gitlab.eclipse.lsp.messages

import org.eclipse.lsp4j.Position
import org.eclipse.lsp4j.TextDocumentIdentifier
import org.eclipse.lsp4j.TextDocumentPositionAndWorkDoneProgressAndPartialResultParams

data class InlineCompletionParams(
  val textDocumentIdentifier: TextDocumentIdentifier,
  val cursorPosition: Position,
  val context: InlineCompletionContext
) : TextDocumentPositionAndWorkDoneProgressAndPartialResultParams(textDocumentIdentifier, cursorPosition)

data class InlineCompletionContext private constructor(val triggerKind: Int) {
  constructor(triggerKind: InlineCompletionTriggerKind) : this(triggerKind.ordinal)
}

enum class InlineCompletionTriggerKind {
  INVOKED,
  AUTOMATIC
}
